libraries/ESP8266mDNS/ESP8266mDNS.cpp

```diff
@@ -589,12 +589,19 @@ void MDNSResponder::_parsePacket(){
         }
       }
 
-      if (answerType == MDNS_TYPE_A) {
+      else if (answerType == MDNS_TYPE_A) {
         partsCollected |= 0x08;
         for (int i = 0; i < 4; i++) {
           answerIp[i] = _conn_read8();
         }
       }
+      else {
+#ifdef MDNS_DEBUG_RX
+          Serial.printf("Ignoring unsupported type\n", tmp8);
```

Comment:
You are passing a parameter to printf, but there isn no corresponding format specifier in format string.